Client/Frontend/Browser/BackForwardTableViewCell.swift

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import Storage
+
+class BackForwardTableViewCell: UITableViewCell {
+    
+    struct BackForwardViewCellUX {
+        static let bgColor = UIColor(colorLiteralRed: 0.7, green: 0.7, blue: 0.7, alpha: 1)
+        static let faviconWidth = 20
+        static let faviconPadding:CGFloat = 20
+        static let labelPadding = 20
+        static let borderSmall = 2
+        static let borderBold = 5
+        static let fontSize:CGFloat = 12.0
+    }
+    
+    lazy var faviconView = UIImageView(image: FaviconFetcher.defaultFavicon)
+    lazy var label = UILabel(frame: CGRectZero)
+    lazy var bg = UIView(frame: CGRect.zero)
+    
+    var connectingForwards = true
+    var connectingBackwards = true
+    
+    var isCurrentTab = false  {
+        didSet {
+            if(isCurrentTab) {
+                label.font = UIFont(name:"HelveticaNeue-Bold", size: BackForwardViewCellUX.fontSize)
+                bg.snp_updateConstraints { make in
+                    make.height.equalTo(BackForwardViewCellUX.faviconWidth+BackForwardViewCellUX.borderBold)
+                    make.width.equalTo(BackForwardViewCellUX.faviconWidth+BackForwardViewCellUX.borderBold)
+                }
+            }
+        }
+    }
+    
+    var site: Site? {
+        didSet {
+            if let s = site {
+                faviconView.setIcon(s.icon, withPlaceholder: FaviconFetcher.defaultFavicon)
+                label.text = s.title
+                setNeedsLayout()
+            }
+        }
+    }
+    
+    var isPrivate = false  {
+        didSet {
+            label.textColor = isPrivate ? UIColor.whiteColor() : UIColor.blackColor()
+        }
+    }
+    
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        backgroundColor = UIColor.clearColor()
+        selectionStyle = .None
+        
+        bg.backgroundColor = BackForwardViewCellUX.bgColor
+        contentView.addSubview(bg)
+        
+        faviconView.backgroundColor = UIColor.whiteColor()
+        contentView.addSubview(faviconView)
+        
+        label.text = " "
+        label.font = label.font.fontWithSize(BackForwardViewCellUX.fontSize)
```

Comment:
Minor nit: Now that label/images are loaded, you can move the code that setups the properties on them inside a lazy initializer:

```swift
lazy var label: UILabel = {
    let label = UILabel()
    label.text = " "
    label.font = label.font.fontWithSize(BackForwardViewCellUX.fontSize)
    return label
}()
```